src/installer/corehost/cli/hostpolicy/static/coreclr_resolver_t.cpp

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <cassert>
+#include <error_codes.h>
+#include <utils.h>
+#include "coreclr_resolver_t.h"
+#include <pal.h>
+#include <trace.h>
+
+extern "C"
+{
+    pal::hresult_t STDMETHODCALLTYPE coreclr_initialize(
+        const char* exePath,
+        const char* appDomainFriendlyName,
+        int propertyCount,
+        const char** propertyKeys,
+        const char** propertyValues,
+        host_handle_t* hostHandle,
+        unsigned int* domainId);
+
+    pal::hresult_t STDMETHODCALLTYPE coreclr_shutdown_2(
+        host_handle_t hostHandle,
+        unsigned int domainId,
+        int* latchedExitCode);
+
+    pal::hresult_t STDMETHODCALLTYPE coreclr_execute_assembly(
+        host_handle_t hostHandle,
+        unsigned int domainId,
+        int argc,
+        const char** argv,
+        const char* managedAssemblyPath,
+        unsigned int* exitCode);
+
+    pal::hresult_t STDMETHODCALLTYPE coreclr_create_delegate(
+        host_handle_t hostHandle,
+        unsigned int domainId,
+        const char* entryPointAssemblyName,
+        const char* entryPointTypeName,
+        const char* entryPointMethodName,
+        void** delegate);
+}
+
+
+#if defined(_WIN32)
+
+bool coreclr_resolver_t::resolve_coreclr(const pal::string_t& libcoreclr_path, coreclr_resolver_contract_t& coreclr_resolver_contract)
+{
+    pal::string_t coreclr_dll_path(libcoreclr_path);
+    append_path(&coreclr_dll_path, LIBCORECLR_NAME);
+
+    pal::dll_t g_coreclr = nullptr;
```

Comment:
Nit: this should not be called `g_...` since it's not a global variable anymore.

Anyway - we should store the library handle in the contract struct (and then store that struct in hostpolicy). If only for debugging purposes, but also this may come in handy in the future so that we can potentially unload it.

Reply:
Agreed. Just tried to not change beyond what is necessary.